labgrid/driver/sshdriver.py

```diff
@@ -111,12 +110,19 @@ def _start_own_master_once(self, timeout):
             ]
 
         env = os.environ.copy()
+        pass_file = ''
         if self.networkservice.password:
-            env['SSHPASS'] = self.networkservice.password
+            fd, pass_file = tempfile.mkstemp()
+            os.fchmod(fd, stat.S_IRWXU)
+            env = {'SSH_ASKPASS': pass_file, 'DISPLAY':'0:0'}
+            with open(fd, 'w') as f:
+                f.write("echo " + self.networkservice.password)
+
         self.process = subprocess.Popen(args, env=env,
                                         stdout=subprocess.PIPE,
                                         stderr=subprocess.STDOUT,
-                                        stdin=subprocess.DEVNULL)
+                                        stdin=subprocess.DEVNULL,
+                                        preexec_fn=os.setsid)
```

Comment:
Why do we need setsid here? Doesn't this risk leaving a process behind?

Reply:
I believe in the pre 8.4 versions they do an extra check that the ssh call is detached from tty.
I'm not sure, I see four ssh processes in htop (even in the new 8.4 version with SSH_ASKPASS_REQUIRE=force and without setsid). How does it look like when you use sshpass and proxy?
![Untitled](https://user-images.githubusercontent.com/36454326/108758912-f64a6600-754b-11eb-8baf-ce3c4db59602.png)